modules/stochastic_tools/test/src/postprocessors/OptimizationTestFunction.C

```diff
@@ -0,0 +1,66 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "OptimizationTestFunction.h"
+
+registerMooseObject("StochasticToolsTestApp", OptimizationTestFunction);
+
+InputParameters
+OptimizationTestFunction::validParams()
+{
+  InputParameters params = GeneralPostprocessor::validParams();
+
+  MooseEnum funcs("rosen eggholder");
+  params.addRequiredParam<MooseEnum>("function", funcs, "Type of function to evaluate.");
+  params.addRequiredParam<std::vector<Real>>("x", "Input values.");
+  params.declareControllable("x");
+  return params;
+}
+
+OptimizationTestFunction::OptimizationTestFunction(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters),
+    _function(getParam<MooseEnum>("function")),
+    _x(getParam<std::vector<Real>>("x"))
+{
+}
+
+PostprocessorValue
+OptimizationTestFunction::getValue() const
+{
+  if (_function == "rosen")
+    return rosen(_x);
+  else if (_function == "eggholder")
+    return eggholder(_x);
+  else
+    paramError("function", "Unknown function: ", _function);
+
+  return 0.0;
+}
+
+Real
+OptimizationTestFunction::rosen(const std::vector<Real> & x)
+{
+  if (x.size() < 2)
+    return 0.0;
+
+  PostprocessorValue sum = 0.0;
+  for (const auto i : make_range(x.size() - 1))
+    sum += 100.0 * Utility::pow<2>(x[i + 1] - x[i] * x[i]) + Utility::pow<2>(1 - x[i]);
```

Comment:
No `std::accumulate`? :D 

Also:

```
for (const auto i : index_range(x))
```

instead

Reply:
Is there a way to `accumulate` when the function considers two neighboring elements? Also, the loop is from 1 to N-1, so I don't think `index_range` can be used unless I break on the last element.